### ISSUE TITLE
Demo project does not work after building with instructions

### ISSUE THREAD

**Issue by aaronfranke** [godotengine/godot-git-plugin#44]
I built this plugin using [the instructions in the README](https://github.com/godotengine/godot-git-plugin#linux):

> ### Linux
> 
> 1. Prepare script for execution: `chmod 755 build_libs.sh`
> 2. Run `. ./build_libs.sh Release`.
> 3. Run `scons platform=x11 target=release`.

While on step 2, I noticed this message:

```
[100%] Built target git2
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
scons: Reading SConscript files ...
```

Looking at the code, it seems that this is caused by `apt-get install scons`. This script certainly needs improvement, it should not try to install scons if it is already installed, and if scons does need to be installed then it's better to ask for root access or inform the user rather than try to run `apt-get` without root. Anyway, this isn't the cause of the main problem in this issue, since on my system `scons` is already installed.

After building, when opening the demo, this is printed to the console:

```
 Can't open dynamic library: /home/aaronfranke/workspace/godot-git-plugin/demo/bin/x11/release/libgitapi.so. Error: /home/aaronfranke/workspace/godot-git-plugin/demo/bin/x11/release/libgitapi.so: undefined symbol: SSL_CTX_free
 modules/gdnative/gdnative.cpp:483 - No valid library handle, can't get symbol from GDNative object
 modules/gdnative/nativescript/nativescript.cpp:1506 - No nativescript_init in "res://bin/x11/release/libgitapi.so" found
```

When trying to run the project (by pressing F5 or the play button), this error is printed in the debugger:

```
E 0:00:00.674   open_dynamic_library: Can't open dynamic library: /home/aaronfranke/workspace/godot-git-plugin/demo/bin/x11/release/libgitapi.so. Error: /home/aaronfranke/workspace/godot-git-plugin/demo/bin/x11/release/libgitapi.so: undefined symbol: SSL_CTX_free
  <C++ Error>   Condition "!p_library_handle" is true. Returned: ERR_CANT_OPEN
  <C++ Source>  drivers/unix/os_unix.cpp:424 @ open_dynamic_library()
```

This problem only happens when building, the 1.2 release build here works fine: https://github.com/godotengine/godot-git-plugin/releases

I'm using Godot 3.2.2 stable on Mint 20 (Ubuntu 20.04). I tested the `master` branch of this repo, the `v1.2` tag, and the `libgit2-bump-fix` branch, all fail in the same way.

**Comment by Janglee123** [comment]
The first issue is known #28. You can remove that line from `build_lib.sh`. The second error `undefined symbol: SSL_CTX_free` is because required libraries are not linked at compile time and also solved in #37.

**Comment by IronicallySerious** [comment]
We found out the importing errors were related to libgit2 dependencies not being linked into the plugin during build time, so @Janglee123 has already fixed that in his PR he linked above.

About the script, we have decided to just let the user install scons on their own and let the script handle the rest. I think that would be enough for building without problems?

**Comment by IronicallySerious** [comment]
@aaronfranke Requesting you to try again with the new `v1` branch. We have done some fixes to the build scripts, build config, and some stability fixes in the code itself with v1.

Sidenote: However #37 hasn't been merged yet in `v1` because that PR is probably only going to work on Godot 4.0, so #37 is going to be merged into `master` instead and the important changes were copied into `v1` branch

**Comment by IronicallySerious** [comment]
The problems in the description have been resolved as far as I have tested. The build script was redone and import issues were resolved and backported from janglee's Godot 4.0 compatible branch into the master branch (which currently is to be seen as the Godot 3.2 compatible branch).

Closing issue.
